### DISCUSSION TITLE
2.0 and roadmap

### DISCUSSION THREAD

**Discussion by vb8448** [bigskysoftware/htmx#2198] [General]
Hi,
is there any public roadmap to htmx 2.0 and the features and changes this release will introduce?

I saw there is a 2.0 branch but hasn't been updated for 6 months.

**Comment by vb8448** [comment]
any update from maintainers? Thanks

**Reply by MikeMoolenaar** [reply]
Carson posted about it yesterday:  https://twitter.com/htmx_org/status/1749837084028960984.
They are working on it in the [v2.0v2.0 branch](https://github.com/bigskysoftware/htmx/tree/v2.0v2.0)

**Comment by Telroshan** [comment]
Hi, unfortunately we don't have an official roadmap to share, but we're hoping to release a htmx 2 beta early March.
Please keep in mind that it's not a promise but a hope though ;)
As mentioned above, you can find some mentions of v2 on Twitter or the Discord. Basically, you won't notice _that_ many changes, most of them were internal changes that users won't even notice. Usage of FormData to properly handle web components, removed deprecated attributes, changed a few config defaults, extensions will move out of the main repo + new extensions API, JSDoc & fixed typescript definition, and some other stuff that I may forget as I'm typing this.
Anyway, there'll be a full changelog when it's out!

**Reply by renannprado** [reply]
@Telroshan looking forward for the improved support to web components ❤️ .
